public/app/features/canvas/runtime/element.tsx

```diff
@@ -432,16 +474,36 @@ export class ElementState implements LayerElement {
     event.target.style.transform = event.transform;
   };
 
+  applyRotate = (event: OnRotate) => {
+    const absoluteRotationDegree = event.absoluteRotation;
+
+    const placement = this.options.placement!;
+    // Ensure rotation is between 0 and 360
+    placement.rotation = absoluteRotationDegree - Math.floor(absoluteRotationDegree / 360) * 360;
+    event.target.style.transform = event.transform;
+  };
+
   // kinda like:
   // https://github.com/grafana/grafana-edge-app/blob/main/src/panels/draw/WrapItem.tsx#L44
   applyResize = (event: OnResize, transformScale = 1) => {
     const placement = this.options.placement!;
 
     const style = event.target.style;
-    const deltaX = event.delta[0] / transformScale;
-    const deltaY = event.delta[1] / transformScale;
-    const dirLR = event.direction[0];
-    const dirTB = event.direction[1];
+    let deltaX = event.delta[0] / transformScale;
+    let deltaY = event.delta[1] / transformScale;
+    let dirLR = event.direction[0];
+    let dirTB = event.direction[1];
+
+    // Handle case when element is rotated
+    if (placement.rotation) {
+      const rotation = placement.rotation ?? 0;
+      const rotationInRadians = (rotation * Math.PI) / 180;
+      const originalDirLR = dirLR;
+      const originalDirTB = dirTB;
+
+      dirLR = Math.sign(originalDirLR * Math.cos(rotationInRadians) - originalDirTB * Math.sin(rotationInRadians));
+      dirTB = Math.sign(originalDirLR * Math.sin(rotationInRadians) + originalDirTB * Math.cos(rotationInRadians));
```

Comment:
and you thought you were done with geometry in school? 🎒

Reply:
SOHCAHTOA!!!!